Source/WebCore/PAL/pal/system/ios/UserInterfaceIdiom.mm

```diff
@@ -28,32 +28,34 @@
 
 #if PLATFORM(IOS_FAMILY)
 
-#import "UIKitSPI.h"
-#import <WebCore/Device.h>
+#import "Device.h"
+#import <pal/spi/ios/UIKitSPI.h>
 
-namespace WebKit {
+#import <pal/ios/UIKitSoftLink.h>
 
-static std::optional<UserInterfaceIdiom> s_currentUserInterfaceIdiom;
+namespace PAL {
+
+static std::atomic<std::optional<UserInterfaceIdiom>> s_currentUserInterfaceIdiom;
```

Comment:
Would be great to check with folks that this does what we hope it does.

Reply:
Abrar and I had a thread about WTF::Atomic, and it's just dealing with a case with swapping that isn't applicable here.

Reply:
I am not so familiar with this code but instead of using `atomic<optional<UserInterfaceIdiom>>` couldn't you use `atomic<UserInterfaceIdiom>` and default `s_currentUserInterfaceIdiom = Unknown` (where Unknown is a new value in the enum) and then instead of checking `!s_.load()` check `s_.load() != Unknown`? Or if we don't want to expand `UserInterfaceIdiom` then we could use `atomic<someIntegerType>` and perform conversions?

Reply:
We've talked with the CPP folks, and this usage is fine.